src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -0,0 +1,367 @@
+        /// <param name="normalize">Whether the feature contributions should be normalized to the [-1, 1] interval.</param>
+        public FeatureContributionCalculatingTransformer(IHostEnvironment env, ICalculateFeatureContribution predictor,
+            string featureColumn = DefaultColumnNames.Features,
+            int top = FeatureContributionCalculatingEstimator.Defaults.Top,
+            int bottom = FeatureContributionCalculatingEstimator.Defaults.Bottom,
+            bool normalize = FeatureContributionCalculatingEstimator.Defaults.Normalize)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(FeatureContributionCalculatingTransformer)))
+        {
+            Host.CheckValue(predictor, nameof(predictor));
+            Host.CheckNonEmpty(featureColumn, nameof(featureColumn));
+            if (top < 0)
+                throw Host.Except($"Number of top contribution must be non negative");
+            if (bottom < 0)
+                throw Host.Except($"Number of bottom contribution must be non negative");
+
+            // If a predictor implements ICalculateFeatureContribution, it also implements the internal interface IFeatureContributionMapper.
+            // This is how we keep the implementation of feature contribution calculation internal.
+            _predictor = predictor as IFeatureContributionMapper;
+            Host.AssertValue(_predictor);
+
+            FeatureColumn = featureColumn;
+            Top = top;
+            Bottom = bottom;
+            Normalize = normalize;
+        }
+
+        // Factory constructor for SignatureLoadModel
+        private FeatureContributionCalculatingTransformer(IHostEnvironment env, ModelLoadContext ctx)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(FeatureContributionCalculatingTransformer)))
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            // *** Binary format ***
+            // IFeatureContributionMapper: predictor
+            // string: featureColumn
+            // int: top
+            // int: bottom
+            // bool: normalize
+
+            ctx.LoadModel<IFeatureContributionMapper, SignatureLoadModel>(env, out _predictor, ModelFileUtils.DirPredictor);
+            FeatureColumn = ctx.LoadNonEmptyString();
+            Top = ctx.Reader.ReadInt32();
+            Contracts.CheckDecode(0 <= Top);
+            Bottom = ctx.Reader.ReadInt32();
+            Contracts.CheckDecode(0 <= Bottom);
+            Normalize = ctx.Reader.ReadBoolByte();
+        }
+
+        // Factory method for SignatureLoadRowMapper.
+        internal static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, Schema inputSchema)
+            => new FeatureContributionCalculatingTransformer(env, ctx).MakeRowMapper(inputSchema);
+
+        // Factory method for Entrypoints.
```

Comment:
>Entrypoints [](start = 30, length = 11)

Is it really needed for entry points? I suspect it's for command line, and we don't want this transformer to be used in command line, right?